### PR TITLE
[WIP] Attempt to make `optimas` generators compatible with generator standard

### PR DESCRIPTION
**Pull request by RemiLehe**
Overall approach:

`optimas` generators currently call the `ask` and `tell` methods, which, under the hood, can the `_ask` and `_tell` methods.

In this PR:
- `ask`/`tell` are renamed to `ask_trials`/`tell_trials` and keep the same interface (for compatibility with the rest of the framework)
- `_ask`/`_tell` are renamed to `ask`/`tell` and adopt the new standardized interface.

This allows other packages to use the `optimas` generators. (In this case, these other packages will call the new, standardized `ask`/`tell` directly, without going through `ask_trials`/`tell_trials`.)

In order to wrap a standardized generator from another package, one would only need to make a thin wrapper around it that defines `ask_trials` and `tell_trials`.

### CODE REVIEW

**Comment by shuds13** [comment]
Looking at this, I think this approach is good. Our current [interfacer](https://github.com/optimas-org/optimas/pull/209/files#diff-bf78501d96dfaddcb21d5691ad8135bd4045f081ff35623d097708edc6ef8c56) in Optimas branch #209 used _ask and _tell as we did not want to do the work with Optimas data structures. Now you have moved that outside to ask_trials / tell_trials so this essentially is the same sort of wrapping but instead of ask - > _ask it is ask_trials -> ask. 

And the ask can use our gens.

File: optimas/generators/base.py [hunk 1]
```diff
@@ -250,7 +253,19 @@ def tell(
...
-        self._tell(trials)
+        # Feed data to the generator, using the standardized API
+        results = []
+        for trial in trials:
+            # Convert to dictionaries
+            point = {
+                **trial.parameters_as_dict(),
+                **trial.analyzed_parameters_as_dict(),
+                **trial.objectives_as_dict(),
+            }
```
Comment (RemiLehe) [inline optimas/generators/base.py:264]: We need to also pass the `id` here.

File: optimas/core/trial.py [hunk 2]
```diff
@@ -81,6 +81,69 @@ def __init__(
             self._mapped_evaluations[ev.parameter.name] = ev
         self.mark_as(TrialStatus.CANDIDATE)
 
+    def to_dict(self) -> Dict:
+        """Convert the trial to a dictionary."""
+        trial_dict = {
+            **self.parameters_as_dict(),
+            **self.objectives_as_dict(),
+            **self.analyzed_parameters_as_dict(),
```
Comment (RemiLehe) [inline optimas/core/trial.py:89]:
```suggestion
            **self.analyzed_parameters_as_dict(),
            **self.custom_parameters_as_dict(),
```

**Comment by RemiLehe** [comment]
Superseeded by https://github.com/optimas-org/optimas/pull/249

**Pull request closed without merging**